gnnflow/cache/cache.py

```diff
@@ -149,29 +149,53 @@ def init_cache(self, *args, **kwargs):
         Init the cache with features
         """
         if self.distributed:
-            return
-
-        if self.dim_node_feat != 0:
-            cache_node_id = torch.arange(
-                self.node_capacity, dtype=torch.int64, device=self.device)
-
-            # Init parameters related to feature fetching
-            self.cache_node_buffer[cache_node_id] = self.node_feats[:self.node_capacity].to(
-                self.device, non_blocking=True)
-            self.cache_node_flag[cache_node_id] = True
-            self.cache_index_to_node_id = cache_node_id
-            self.cache_node_map[cache_node_id] = cache_node_id
-
-        if self.dim_edge_feat != 0:
-            cache_edge_id = torch.arange(
-                self.edge_capacity, dtype=torch.int64, device=self.device)
-
-            # Init parameters related to feature fetching
-            self.cache_edge_buffer[cache_edge_id] = self.edge_feats[:self.edge_capacity].to(
-                self.device, non_blocking=True)
-            self.cache_edge_flag[cache_edge_id] = True
-            self.cache_index_to_edge_id = cache_edge_id
-            self.cache_edge_map[cache_edge_id] = cache_edge_id
+            # the edge map is ordered my insertion order, which is the order of ts
+            if self.dim_edge_feat != 0:
+                keys, feats = self.kvstore_client.init_cache(
+                    self.edge_capacity)
+                if len(keys) >= self.edge_capacity:
```

Comment:
Why `len(keys)` can be greater than `self.edge_capacity` when you pass `self.edge_capacity` as argument in the `kvstore_client.init_cache`?

Reply:
It's a typo here. There are only 2 cases. Len(keys) == edge_capacity and len(keys) < edge_capacity.

Reply:
Why we need to distinguish these two cases?